src/Firebase/Valinor/Source.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase\Valinor;
+
+use CuyZ\Valinor\Mapper\Source\Source as BaseSource;
+use IteratorAggregate;
+use Kreait\Firebase\Exception\InvalidArgumentException;
+use SplFileObject;
+use Throwable;
+use Traversable;
+
+/**
+ * @internal
+ *
+ * @implements IteratorAggregate<mixed>
+ */
+final class Source implements IteratorAggregate
+{
+    private function __construct(
+        /** @var iterable<mixed> */
+        private readonly iterable $delegate,
+    ) {
+    }
+
+    public static function parse(mixed $value): self
+    {
+        if (is_iterable($value)) {
+            return new self(BaseSource::iterable($value));
+        }
+
+        if (str_starts_with((string) $value, '{')) {
+            try {
+                return new self(BaseSource::json($value));
+            } catch (Throwable $e) {
+                throw new InvalidArgumentException(message: $e->getMessage(), previous: $e);
+            }
+        }
+
+        try {
+            return new self(BaseSource::file(new SplFileObject($value)));
```

Comment:
Just want to highlight that this was a small breaking change for us, because files are now expected to have a `.json`, `.yaml` or `.yml` extension, and this wasn't required before. Maybe an edge case, because of the way we provision config in our projects. We ended up simply reading the file ourselves before passing it to the lib.

Reply:
I'm addressing this in https://github.com/kreait/firebase-php/pull/1013 - I hope this will restore the previous behavior for you! 🤞🏻